src/devices/llama_devices.ts

```diff
@@ -0,0 +1,150 @@
+import {Definition, Fz, ModernExtend} from '../lib/types';
+import * as exposes from '../lib/exposes';
+import * as modernExtend from '../lib/modernExtend';
+
+const e = exposes.presets;
+const prefixEnpointName: string = 'button_';
+const endpointCount = 4;
+
+type Config = {
+    attribute: string,
+    name: string,
+    description: string,
+    values: ConfigValues[]
+};
+type ConfigValues = {
+    name: string,
+    value: number,
+    command?: string
+};
+const switchTypeConf: Config = {
+    attribute: 'switchType',
+    name: 'switch_type',
+    description: 'Operation method of the switch',
+    values: [
+        {
+            name: 'toggle',
+            value: 0,
+        },
+        // {
+        //     name: 'momentary',
+        //     value: 1
+        // },
+        {
+            name: 'alternate',
+            value: 2,
+        },
+    ],
+};
+const switchAtionConf: Config = {
+    attribute: 'switchActions',
+    name: 'switch_actions',
+    description: 'Command to be generated when the switch moves between its two states',
+    values: [
+        {
+            name: 'on/off',
+            value: 0,
+        },
+        {
+            name: 'off/on',
+            value: 1,
+        },
+        {
+            name: 'toggle',
+            value: 2,
+        },
+    ],
+};
+const configs: Config[] = [
+    switchTypeConf,
+    switchAtionConf,
+];
+
```

Comment:
All these functions make the code hard to read, I propose to just do:

```
const switchCfg = (epName) =>  modernExtend.enumLookup({
        name: cfg.name,
        lookup: mapObject(cfg.values, (k) => k.name, (v) => v.value),
        endpointName: epName,
        cluster: 'genOnOffSwitchCfg',
        attribute: cfg.attribute,
        description: cfg.description,
        access: 'ALL',
        entityCategory: 'config',
    })
```

Remove all the `cfg.` and add them directly there.

Reply:
I use values in configs in several places. you want me hardcode same values in different places?

Reply:
done